components/sync15/src/request.rs

```diff
@@ -142,13 +151,14 @@ pub struct UploadResult {
 
 pub type PostResponse = Sync15ClientResponse<UploadResult>;
 
+//REMOVE BatchState Unnsupported for issue 1144
```

Comment:
Rm this comment. The enum itself does make sense: uploading batches is stateful, so we do need to know whether we are in a batch or not.

Consider renaming `NoBatch` to `NewBatch`?